docs/overrides/main.html

```diff
@@ -3,5 +3,87 @@
+        var searchForm = document.querySelector('.md-search__form');
+        var searchInput = document.querySelector('[data-md-component="search-query"]');
+        var searchResults = document.querySelector('.md-search__output');
+        var lastSearchTerm = '';
+        var searchTimeout;
+        var lastSearchTime = 0;
+        const DEBOUNCE_MS = 1000; // Wait 1 second between typed searches
+
+        if (searchForm && searchInput && searchResults) {
+            // Track when search results appear from typing (debounced)
+            searchInput.addEventListener('input', function() {
+                clearTimeout(searchTimeout);
+                const searchTerm = this.value.trim();
+
+                // Don't track if term hasn't changed or is too short
+                if (searchTerm === lastSearchTerm || searchTerm.length < 3) {
+                    return;
+                }
+
+                searchTimeout = setTimeout(function() {
+                    const now = Date.now();
+                    // Only track if:
+                    // 1. Results exist
+                    // 2. Enough time has passed since last search
+                    if (searchResults.querySelector('.md-search-result__item') &&
+                        now - lastSearchTime > DEBOUNCE_MS) {
+                        posthog.capture('documentation_search', {
+                            search_term: searchTerm,
+                            search_type: 'typed'
+                        });
+                        lastSearchTerm = searchTerm;
+                        lastSearchTime = now;
+                    }
+                }, DEBOUNCE_MS);
+            });
+
+            // Track submitted searches
+            searchForm.addEventListener('submit', function(e) {
+                e.preventDefault();
+                clearTimeout(searchTimeout); // Cancel any pending typed search
+                const searchTerm = searchInput.value.trim();
+
+                // Only track if:
+                // 1. Term exists
+                // 2. Term is different from last search OR it's been a while
+                const now = Date.now();
+                if (searchTerm &&
+                    (searchTerm !== lastSearchTerm || now - lastSearchTime > DEBOUNCE_MS)) {
+                    posthog.capture('documentation_search', {
+                        search_term: searchTerm,
+                        search_type: 'submitted'
+                    });
+                    lastSearchTerm = searchTerm;
+                    lastSearchTime = now;
+                }
+            });
+
+            // Track result clicks (unchanged)
+            searchResults.addEventListener('click', function(e) {
+                var clickedLink = e.target.closest('a');
+                if (clickedLink) {
+                    posthog.capture('search_result_click', {
+                        search_term: searchInput.value.trim(),
+                        clicked_url: clickedLink.href,
```

Comment:
If I search for `foo` then the links include `?h=foo` at the end for highlighting, probably want to trim that so that we can sensibly group by the search result URLs.

Also I suggest splitting out the URL fragment as another attribute.

Reply:
Done